components/challenges/TimerClock.js

```diff
@@ -1,28 +1,70 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { View } from "react-native";
 import ProgressCircle from 'react-native-progress-circle';
+import moment from 'moment';
+import KitText from "../KitText"
+import Rectangle from "./Rectangle";
 
+import { getUserId } from "../../utils/auth/auth";
 import Colors from "../../constants/Colors";
 
-import KitText from "../KitText"
 const diameter = 96;
 export default function TimerClock(props) {
+  const [expireMoment, setExpireMoment] = useState(null);
+  const [minGiven, setMinGiven] = useState(null);
+
+  //EVAN TODO: Verify that the math behind all this works, and we don't need to adjust for timezone in some way
+  useEffect(() => {
+    const currentUser = getUserId();
+    const openedTimeForUser = props.challenge.opened[currentUser]
+    const minsGivenForChallenge = props.challenge.challengeDetails.minGiven;
+    if (!minsGivenForChallenge || !openedTimeForUser) return; //Should instead do error handling here, but TimerClock should never be used without passing a challenge to it
```

Comment:
I think if you want to check this error, you also have to check if currentUser is a property of opened. 
```if props.challenge.opened.hasOwnProperty(currentUser)```

Reply:
Shouldn't the `!openedTimeForUser` check for this? because the way `openedTimeForUser` is declared it should be undefined if `props.challenge.opened` does not have the current user and `!undefined` should be truthy

Reply:
My bad, I thought it threw a invalid key error or something